agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/connectionstring/ConnectionString.java

```diff
@@ -0,0 +1,192 @@
+
+  public abstract URL getStatsbeatEndpoint();
+
+  private static Map<String, String> getKeyValuePairs(String connectionString) {
+    if (connectionString.length() > CONNECTION_STRING_MAX_LENGTH) { // guard against malicious input
+      throw new IllegalArgumentException(
+          "ConnectionString values with more than "
+              + CONNECTION_STRING_MAX_LENGTH
+              + " characters are not allowed.");
+    }
+    // parse key value pairs
+    Map<String, String> kvps;
+    try {
+      kvps = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+      kvps.putAll(extractKeyValuesFromConnectionString(connectionString));
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Could not parse connection string.", e);
+    }
+
+    return kvps;
+  }
+
+  private static Map<String, String> extractKeyValuesFromConnectionString(String connectionString) {
+    Objects.requireNonNull(connectionString);
+    Map<String, String> keyValues = new HashMap<>();
+    String[] splits = connectionString.split(";");
+    for (String split : splits) {
+      String[] keyValPair = split.split("=");
+      if (keyValPair.length == 2) {
+        keyValues.put(keyValPair[0], keyValPair[1]);
+      }
+    }
+    return keyValues;
+  }
+
+  private static ConnectionString mapToConnectionConfiguration(
+      Map<String, String> kvps,
+      @Nullable String statsbeatInstrumentationKey,
+      @Nullable String statsbeatIngestionEndpoint) {
+
+    // get ikey
+    String instrumentationKey = kvps.get(Keywords.INSTRUMENTATION_KEY);
+    if (CoreUtils.isNullOrEmpty(instrumentationKey)) {
+      throw new IllegalArgumentException("Missing '" + Keywords.INSTRUMENTATION_KEY + "'");
+    }
+
+    ConnectionStringBuilder endpoints = new ConnectionStringBuilder();
+
+    // resolve suffix
+    String suffix = kvps.get(Keywords.ENDPOINT_SUFFIX);
+    if (!CoreUtils.isNullOrEmpty(suffix)) {
+      if (suffix.startsWith(".")) {
+        suffix = suffix.substring(1);
+      }
+      endpoints.setIngestionEndpoint(
+          "https://" + EndpointPrefixes.INGESTION_ENDPOINT_PREFIX + "." + suffix);
+      endpoints.setLiveEndpoint("https://" + EndpointPrefixes.LIVE_ENDPOINT_PREFIX + "." + suffix);
+      endpoints.setProfilerEndpoint(
+          "https://" + EndpointPrefixes.PROFILER_ENDPOINT_PREFIX + "." + suffix);
+      endpoints.setSnapshotEndpoint(
+          "https://" + EndpointPrefixes.SNAPSHOT_ENDPOINT_PREFIX + "." + suffix);
+    }
+
+    // set explicit endpoints
+    String liveEndpoint = kvps.get(Keywords.LIVE_ENDPOINT);
+    if (!CoreUtils.isNullOrEmpty(liveEndpoint)) {
+      endpoints.setLiveEndpoint(liveEndpoint);
+    }
+
+    String ingestionEndpoint = kvps.get(Keywords.INGESTION_ENDPOINT);
+    if (!CoreUtils.isNullOrEmpty(ingestionEndpoint)) {
+      endpoints.setIngestionEndpoint(ingestionEndpoint);
+    }
+
+    String profilerEndpoint = kvps.get(Keywords.PROFILER_ENDPOINT);
```

Comment:
additionally, statsbeat has nothing to do with profile/livemetric/snapshot endpoint.  i think it's better to separate it out.